azurerm/internal/services/compute/virtual_machine_scale_set_import.go

```diff
@@ -96,6 +98,21 @@ func importVirtualMachineScaleSet(osType compute.OperatingSystemTypes, resourceT
 			d.Set("admin_password", "ignored-as-imported")
 		}
 
+		if features.VMSSExtensionsBeta() {
+			if vm.VirtualMachineScaleSetProperties.VirtualMachineProfile.ExtensionProfile != nil {
+				if extensionsProfile := vm.VirtualMachineScaleSetProperties.VirtualMachineProfile.ExtensionProfile; extensionsProfile != nil {
+					for _, v := range *extensionsProfile.Extensions {
+						v.ProtectedSettings = ""
+					}
+					updatedExtensions, err := flattenVirtualMachineScaleSetExtensions(extensionsProfile, d)
+					if err != nil {
+						return []*schema.ResourceData{}, fmt.Errorf("could not read VMSS extensions data for %q (resource group %q)", id.Name, id.ResourceGroup)
+					}
+					d.Set("extension", updatedExtensions)
```

Comment:
if the feature's enabled and we're not inside this If,otherwise this wants setting to an empty value?